### PR TITLE
Add MoveTo tests & make them pass

### PR DESCRIPTION
**Pull request by v4hn**
- implement a simple IK interface for `JointInterpolationSolver` Cartesian goals
- fix some bugs
- include attached objects/subframe fixes similar to #296

### CODE REVIEW

**Review by rhaschke — APPROVED** [review]
Looks good. Sorry for the long delay in reviewing this. I focussed on the MoveIt release first.
Note the minor fixup commits. They should be squashed into their base commit before merging (I guess you want to keep history).

**Comment by rhaschke** [comment]
Sorry, I didn't test on Melodic... Please, just drop my latest commit (8408fba17d75efbf7cd5716ba51899ff96cd612b).

**Comment by rhaschke** [comment]
Hm. https://github.com/ros-planning/moveit_resources/pull/92 broke the pick+place test. Should be fixed now.

File: core/src/stages/move_to.cpp [hunk 1]
```diff
@@ -204,47 +202,81 @@ bool MoveTo::compute(const InterfaceState& state, planning_scene::PlanningSceneP
...
+			if (ik_pose_msg.header.frame_id.empty() && !get_tip(ik_pose_msg.header.frame_id)) {
+				solution.markAsFailure("frame_id of ik_frame is empty and no unique group tip was found");
+				return false;
+			} else if (!scene->knowsFrameTransform(ik_pose_msg.header.frame_id)) {
+				std::stringstream ss;
+				ss << "ik_frame specified in unknown frame '" << ik_pose_msg << "'";
+				solution.markAsFailure(ss.str());
 				return false;
 			}
 		}
```
Comment (rhaschke) [inline core/src/stages/move_to.cpp:240]: This block of code is required 1:1 for `MoveRelative` as well:
https://github.com/ros-planning/moveit_task_constructor/blob/7ec874572c3b35f1cdd7a71c5e6b94c749b3093b/core/src/stages/move_relative.cpp#L199-L213

We should factor that out into a function...

File: core/test/test_move_to.cpp [hunk 2]
```diff
@@ -70,29 +72,87 @@ TEST_F(PandaMoveTo, stateTarget) {
...
+	aco.object.header.frame_id = aco.link_name;
+	aco.object.operation = aco.object.ADD;
+	aco.object.id = id;
+	aco.object.primitives.resize(1, []() {
+		shape_msgs::SolidPrimitive p;
+		p.type = p.SPHERE;
+		p.dimensions.resize(1);
+		p.dimensions[p.SPHERE_RADIUS] = 0.01;
+		return p;
+	}());
```
[the remark this answers is not in the document]
Comment (v4hn) [reply core/test/test_move_to.cpp:119]: ROS messages do not have useful constructors, so it's pretty much a replacement for that.
The main advantage is that scope does not leak. In this case the variable `p` does not stay available outside the argument. I use it as a more concise version of extra braces to restrict scope `{ ... }`.

The real advantage of [IIFE](https://rigtorp.se/iife/) (I'm not sure I used that anywhere) is that you can initialize const objects that way too.

> I guess they are inlined automatically by the compiler?

[Yes](https://godbolt.org/z/Krjvrqvnj).

File: core/src/stages/move_to.cpp (hunk 1 above)
Comment (v4hn) [reply core/src/stages/move_to.cpp]: please have another look. I'm happy with the code again (assuming CI is) and I addressed this too.

File: core/test/test_move_to.cpp [hunk 3]
```diff
@@ -62,7 +62,7 @@ TEST_F(PandaMoveTo, mapTarget) {
 }
 
 TEST_F(PandaMoveTo, stateTarget) {
-	move_to->setGoal([]() {
+	move_to->setGoal([] {
```
Comment (rhaschke) [inline core/test/test_move_to.cpp]: Interesting. Haven't seen this syntax before :wink:

**Review by rhaschke — APPROVED** [review]
Thanks for the cleanup. LGTM. Note, that I didn't test myself - I assume you did that :wink:

File: core/test/test_move_to.cpp (hunk 3 above)
Comment (v4hn) [reply core/test/test_move_to.cpp]: Me neither, but it's nice. The downside is that for some reason `[] -> Type {}` is not supported syntax though and requires `[]() -> Type {}` (which reads so much... better?)

**Pull request merged**